ext/filter/filter.c

```diff
@@ -690,29 +690,13 @@ PHP_FUNCTION(filter_input_array)
 	}
 
 	array_input = php_filter_get_storage(fetch_from);
+
 	if (EG(exception)) {
 		RETURN_THROWS();
 	}
 
 	if (!array_input) {
-		zend_long filter_flags = 0;
-		zval *option;
-		if (op_long) {
-			filter_flags = op_long;
-		} else if (op_ht && (option = zend_hash_str_find(op_ht, "flags", sizeof("flags") - 1)) != NULL) {
```

Comment:
What is the problem with the `["flags" => FILTER_NULL_ON_FAILURE]` value? As far as I saw, the other functions look for this key as well

Reply:
According to my understandig, 2 things are going on here.

1) `op_long`
is asking, in the wrong order `if (op_long)`, BUT ALSO and more important after that it "asks" if the flag `FILTER_NULL_ON_FAILURE` is there, doing `filter_flags & FILTER_NULL_ON_FAILURE` , like if we could pass that flag doing something like: `filter_input_array(INPUT_GET, FILTER_DEFAULT|FILTER_NULL_ON_FAILURE)` .... and seems that we are not suppose to pass flags like this because FILTER_DEFAULT is a filter and FILTER_NULL_ON_FAILURE is a flag, semantically 2 different things, and they are suppose to be passed in a separated way. Actually if we do something like that we will get a warning like
`Warning: filter_input_array(): Unknown filter with ID `

2) This part of the code:

```c
} else if (op_ht && (option = zend_hash_str_find(op_ht, "flags", sizeof("flags") - 1)) != NULL) {
```

Works only if in the second parameter we pass something like:

`filter_input_array(INPUT_GET, ["flags" => FILTER_NULL_ON_FAILURE])`

And this is ambiguous, because the array is suppose to have names that should exist in INPUT_GET, so the function could make some filtering on them. For each "key" in the array, we do have the possibility to specify filters, options and flags.

I believe , again, this is wrong, because we cannot say to our developers:

"The second parameter should be an array holding the name of the parameters in $type (first argument) each one of them with filters/option/flags; BUT you cannot expect to have a parameter named "flags" in $type (first argument) because that name is special for this function" ... please let no add more messyness to this :) 


The good news is that if $type is missing (first argument) ... no one, no code, is able to get a FALSE at the moment, is just not possible, so whatever we do will have no impact.

Thanks a lot for the review @kocsismate , sorry for the delay in my response, I will wait for you next response.